### PR TITLE
Find Usages: more J2K and read action removal (KT-7917)

### PR DESCRIPTION
**Pull request by yole**

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/findUsages/handlers/KotlinFindUsagesHandler.kt [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            elementsToSearch.toTypedArray()
+    }
+
+    protected fun searchTextOccurrences(element: PsiElement, processor: Processor<UsageInfo>, options: FindUsagesOptions): Boolean {
+        val scope = options.searchScope
+
+        val searchText = options.isSearchForTextOccurrences && scope is GlobalSearchScope
+
+        if (searchText) {
+            if (options.fastTrack != null) {
```
Comment (VladimirReshetnikov) [inline idea/src/org/jetbrains/kotlin/idea/findUsages/handlers/KotlinFindUsagesHandler.kt]: `options.fastTrack?.searchCustom ...`

File: idea/src/org/jetbrains/kotlin/idea/findUsages/handlers/KotlinFindUsagesHandler.kt [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        return if (elementsToSearch.isEmpty())
+            arrayOf(getPsiElement())
+        else
+            elementsToSearch.toTypedArray()
+    }
+
+    protected fun searchTextOccurrences(element: PsiElement, processor: Processor<UsageInfo>, options: FindUsagesOptions): Boolean {
+        val scope = options.searchScope
+
+        val searchText = options.isSearchForTextOccurrences && scope is GlobalSearchScope
```
Comment (VladimirReshetnikov) [inline idea/src/org/jetbrains/kotlin/idea/findUsages/handlers/KotlinFindUsagesHandler.kt]: If you inline this variable, you can get rid of `as` operators below.

File: idea/src/org/jetbrains/kotlin/idea/findUsages/handlers/KotlinFindUsagesHandler.kt (hunk 1 above)
Comment (VladimirReshetnikov) [inline idea/src/org/jetbrains/kotlin/idea/findUsages/handlers/KotlinFindUsagesHandler.kt]: If you revert this condition, you can get rid of `else`.

**Comment by yole** [comment]
Merged manually.

**Pull request closed without merging**